forest/jasmine/traj2stats.py

```diff
@@ -819,121 +824,290 @@ def gps_quality_check(study_folder: str, study_id: str) -> float:
+                            f"{output_folder}/trajectory/{id_code}.csv"
+                        )
+                        pd_traj.to_csv(dest_path, index=False)
+                    if frequency == Frequency.BOTH:
+                        summary_stats1, logs1 = gps_summaries(
+                            traj,
+                            tz_str,
+                            Frequency.HOURLY,
+                            places_of_interest,
+                            save_log,
+                            threshold,
+                            split_day_night,
+                        )
+                        write_all_summaries(
+                            id_code, summary_stats1, f"{output_folder}/hourly"
+                        )
+                        summary_stats2, logs2 = gps_summaries(
+                            traj,
+                            tz_str,
+                            Frequency.DAILY,
+                            places_of_interest,
+                            save_log,
+                            threshold,
+                            split_day_night,
+                            person_point_radius,
+                            place_point_radius,
+                        )
+                        write_all_summaries(
+                            id_code, summary_stats2, f"{output_folder}/daily"
+                        )
+                        if save_log:
+                            with open(
+                                f"{output_folder}/hourly/locations_logs.json",
+                                "w",
+                            ) as hourly:
+                                json.dump(logs1, hourly, indent=4)
+                            with open(
+                                f"{output_folder}/daily/locations_logs.json",
+                                "w",
+                            ) as daily:
+                                json.dump(logs2, daily, indent=4)
                     else:
-                        summary_stats = gps_summaries(traj,tz_str,option)
-                        write_all_summaries(ID, summary_stats, output_folder)
+                        freq = frequency.value
+                        log_dir = f"{output_folder}/{freq}/locations_logs.json"
+                        summary_stats, logs = gps_summaries(
+                            traj,
+                            tz_str,
+                            frequency,
+                            places_of_interest,
+                            save_log,
+                            threshold,
+                            split_day_night,
+                        )
+                        write_all_summaries(
+                            id_code, summary_stats, output_folder
+                        )
+                        if save_log:
+                            with open(
+                                log_dir,
+                                "w",
+                            ) as loc:
```

Comment:
If a statement can fit on one line then spreading it across multiple lines actually hurts readability.